File: app/src/app/core/services/api/models/sporocilo.ts
import { Kontakt } from './kontakt'
import { Id } from './id';
export interface Sporocilo {
	'_id': Id<Sporocilo>
	kontakt_posiljatelj_id: Id<Kontakt>
	kontakt_prejemnik_id: Array<Id<Kontakt>>
	poslano: string
	vsebina: string
}

File: api/core/src/main/kotlin/core/domain/Sporocilo.kt
package core.domain

import core.base.Encrypted
import core.base.Id
import core.extend.zdaj
import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable

@Serializable
data class Sporocilo(
    override var _id: Id<Sporocilo> = Id(),
    var kontakt_posiljatelj_id: Id<Kontakt>,
    var kontakt_prejemnik_id: MutableSet<Id<Kontakt>>,
    var poslano: LocalDateTime = LocalDateTime.zdaj(),
    @Contextual val vsebina: Encrypted,
) : Entiteta<Sporocilo>
